### PR TITLE
Inference Pipeline Optimization - vectorize get target coords

### PR DESCRIPTION
**Pull request by javak87**
## Description
Target-coordinate encoding spent most of its time **splitting points into one tiny tensor per HEALPix cell** (`12 * 4**hl`), concatenating that list again inside `locs_to_cell_coords_ctrs` / `locs_to_ctr_coords`, and repeating that for 5 vertex frames and 8 neighbor-center frames. This keeps the same output layout and replaces the split/cat round-trips with packed `bmm` / `matmul`.
- **`_rotate_points_per_cell`:** `repeat_interleave` cell ids from `masked_points_per_cell`, then one `torch.bmm` with that cell’s 3×3. Same math as `locs_to_cell_coords_ctrs` after `torch.cat` of the per-cell list, without building the list.
- **`verts_local`:** one `torch.repeat_interleave` on occupied cells instead of a Python `repeat` per cell and `torch.cat`.
- **Eight neighbor frames:** one `vecs_to_rots` on all neighbors×cells and one batched `matmul`, instead of eight `locs_to_ctr_coords` calls (each cat → bmm → split → cat).
- **`tokenize_apply_mask_target`:** unchanged; it still calls `get_target_coords_local` with packed coords and counts.

## Inference timing
**Environment:** Jupiter, 1 node, mini-epoch 8 from `cw6a4szu`
The following command was used for both branches. `test_config.output.num_samples=0` is required to avoid OOM.
```bash
../WeatherGenerator-private/hpc/launch-slurm.py --stage inference --from-run-id cw6a4szu \
  --mini-epoch 8 --nodes 1 --time 15:00 \
  --options test_config.output.num_samples=0 test_config.forecast.num_steps=8 \
    test_config.samples_per_mini_epoch=4 test_config.validation_noise_levels=[] \
    data_loading.num_workers=1 data_loading.num_workers_validation=1 \
    test_config.compute_full_validation_loss=False
```
| Run | RUN_ID | Slurm | s/it | loss | valid? |
| --- | --- | --- | --- | --- | --- |
| 1 | `hauhyu9g` | `1517706` | 36.77 | 3.2941E-02 | yes |
| 2 | `at9hwlw5` | `1517710` | 36.47 | 3.2944E-02 | yes |
| 3 | `qdn8mf8r` | `1517712` | 36.27 | 3.2944E-02 | yes |
| **Mean (valid)** | — | — | **36.50** | cluster | **yes** |

Vs **develop-ssl-diffusion-v1** control mean **59.15 s/it**: **1.62×**.

**Note:** These timings only show up if the job tree is running `javad/optimize-hpy_splits`. `launch-slurm.py --from-run-id cw6a4szu` copies Python from that training run, not from your checkout, so launching from the optimized branch alone does not change `get_target_coords_local` unless those files are overlaid (or otherwise copied) into the new job directory.

## Issue Number
#2778 

Is this PR a draft? Mark it as draft.

## Checklist before asking for review

-   [ ] I have performed a self-review of my code
-   [ ] My changes comply with basic sanity checks:
      - I have fixed formatting issues with `./scripts/actions.sh lint`
      - I have run unit tests with `./scripts/actions.sh unit-test`
      - I have documented my code and I have updated the docstrings.
      - I have added unit tests, if relevant
-   [ ] I have tried my changes with data and code:
      - I have run the integration tests with `./scripts/actions.sh integration-test`
      - (bigger changes) I have run a full training and I have written in the comment the run_id(s): `launch-slurm.py --time 60`
      - (bigger changes and experiments) I have shared a hegdedoc in the github issue with all the configurations and runs for this experiments
-   [ ] I have informed and aligned with people impacted by my change:
    - for config changes: the MatterMost channels and/or a design doc
    - for changes of dependencies: the MatterMost software development channel

#### FastEvaluation
 - [ ] I have updated the public documentation if necessary

### CODE REVIEW

File: src/weathergen/datasets/tokenizer_utils.py [hunk 1]
```diff
@@ -6,10 +6,9 @@
 
 from weathergen.common.io import IOReaderData
 from weathergen.datasets.utils import (
-    locs_to_cell_coords_ctrs,
```
Comment (florianscheidl) [inline src/weathergen/datasets/tokenizer_utils.py:9]: These two functions are not used anywhere else, except for tests. Before merging the PR, we should remove them (and the associated tests).

File: tests/test_get_target_coords_local.py
Comment (florianscheidl) [inline tests/test_get_target_coords_local.py:1]: Similar comment as for another PR: I think it's a good sanity check that the old and new implementations match. In my opinion, a standalone test for a simple case makes more sense.

**Review by florianscheidl — COMMENTED** [review]
Looks good to me! Suggest a standalone test and removing the functions in the dataset utils.

File: src/weathergen/datasets/tokenizer_utils.py (hunk 1 above)
Comment (javak87) [reply src/weathergen/datasets/tokenizer_utils.py:9]: removed!

**Review by florianscheidl — APPROVED** [review]
Lgtm
